From ee4b81cbc232d47ba9940f1987777b17452e71ff Mon Sep 17 00:00:00 2001
From: Jan Tojnar <jtojnar@gmail.com>
Date: Fri, 18 Oct 2019 01:46:19 +0200
Subject: [PATCH] UI: Fix build
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Bustle/UI.hs:490:3: error:
    • No instance for (Control.Monad.Fail.MonadFail
                         (Bustle BConfig BState))
        arising from a do statement
        with the failable pattern ‘[openItem, openTwoItem]’
    • In a stmt of a 'do' block:
        [openItem, openTwoItem] <- mapM
                                     (getW castToMenuItem) ["open", "openTwo"]
      In the expression:
        do builder <- io builderNew
           io
             $ builderAddFromFile builder =<< getDataFileName "data/bustle.ui"
           let getW cast name = io $ builderGetObject builder cast name
           window <- getW castToWindow "diagramWindow"
           ....
      In an equation for ‘emptyWindow’:
          emptyWindow
            = do builder <- io builderNew
                 io
                   $ builderAddFromFile builder =<< getDataFileName "data/bustle.ui"
                 let getW cast name = ...
                 ....
    |
490 |   [openItem, openTwoItem] <- mapM (getW castToMenuItem) ["open", "openTwo"]
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
---
 Bustle/UI.hs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Bustle/UI.hs b/Bustle/UI.hs
index 28ede19..6666c72 100644
--- a/Bustle/UI.hs
+++ b/Bustle/UI.hs
@@ -487,7 +487,8 @@ emptyWindow = do
   subtitle <- getW castToLabel "headerSubtitle"
   spinner  <- getW castToSpinner "headerSpinner"
 
-  [openItem, openTwoItem] <- mapM (getW castToMenuItem) ["open", "openTwo"]
+  openItem <- getW castToMenuItem "open"
+  openTwoItem <- getW castToMenuItem "openTwo"
   recordSessionItem <- getW castToMenuItem "recordSession"
   recordSystemItem <- getW castToMenuItem "recordSystem"
   recordAddressItem <- getW castToMenuItem "recordAddress"
-- 
2.26.2

